### PR TITLE
[libatomic-ops] Update to v7.8.0

### PR DESCRIPTION
**Pull request by ivmai**
Update to latest upstream, major changes are: atomic_ops_stack fix for arm, refined license terms.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by ivmai** [comment]
This PR replaces https://github.com/microsoft/vcpkg/pull/26566 (the canceled one)

File: ports/libatomic-ops/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "libatomic-ops",
-  "version": "7.7.0",
-  "port-version": 1,
+  "version": "7.8.0",
```
Comment (JonLiu1993) [inline ports/libatomic-ops/vcpkg.json:3]:
```suggestion
  "version": "${VERSION}",
```

File: ports/libatomic-ops/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ivmai/libatomic_ops
-    REF 7a8de3bd9c6c61c68a866b849e7b1d17d76d2d36 # v7.7.0-20211109
-    SHA512 05555792a199526d8e164833f590cc57c5ee34672d81952787a09dd7008e947e4e8b6ad63fb6b8ee315294b98fdf743639622b3d9156d8a8f8363b431e875c45
+    REF v7.8.0
```
[the remark this answers is not in the document]
Comment (ivmai) [reply ports/libatomic-ops/portfile.cmake]: Done

**Comment by ivmai** [comment]
Here's an error after the change:
    REF "v7.8.0"
->
    REF "v${VERSION}"

How to fix it better?

File: ports/libatomic-ops/portfile.cmake [hunk 3]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ivmai/libatomic_ops
-    REF 7a8de3bd9c6c61c68a866b849e7b1d17d76d2d36 # v7.7.0-20211109
-    SHA512 05555792a199526d8e164833f590cc57c5ee34672d81952787a09dd7008e947e4e8b6ad63fb6b8ee315294b98fdf743639622b3d9156d8a8f8363b431e875c45
+    REF "v${VERSION}"
```
Comment (JonLiu1993) [inline ports/libatomic-ops/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

Comment (ivmai) [reply ports/libatomic-ops/portfile.cmake:4]: Aha, I probably know the root cause of the failure...

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**